### ISSUE TITLE
Project structure

### ISSUE THREAD

**Issue by jpivarski** [PerilousApricot/spark-ttree#4]
The structure isn't quite right:

```
src
├── main
│   └── java
│       └── edu
│           └── vanderbilt
│               └── accre
│                   ├── array
│                   │   ├── Array.java
│                   │   ├── JaggedArray.java
│                   │   ├── PrimitiveArray.java
│                   │   └── RawArray.java
│                   ├── ArrayBuilder.java
│                   ├── interpretation
│                   │   ├── AsDtype.java
│                   │   └── Interpretation.java
│                   ├── root_proxy
│                   │   ├── Constants.java
│                   │   ├── FileInterface.java
│                   │   ├── HadoopFile.java
│                   │   ├── IOFactory.java
│                   │   ├── NIOFile.java
│                   │   └── RootFile.java
│                   └── spark_ttree
└── test
    └── java
        └── edu
            └── vanderbilt
                └── accre
                    ├── ArrayBuilderTest.java
                    ├── InterpretationsTest.java
                    └── root_proxy
                        └── IOTest.java
```

If the groupId is `edu.vanderbilt.accre` and the artifactId is `spark-ttree`, there should be another directory level in there between `accre` and our code (both in `main` and in `test`). Like this:

```
src
├── main
│   └── java
│       └── edu
│           └── vanderbilt
│               └── accre
│                   └── spark-ttree
│                       ├── ...
```

That directory should be the project name, but `spark-ttree` has a hyphen in it, which can't be used in the `package` line at the top of each file, and that `package` has to match the hierarchy. I don't remember how Spark-ROOT did it, but that was a Scala project, and Scala is more flexible on these things.

@PerilousApricot, how do you feel about underscores instead of hyphens? A cooler name?  :)

Incidentally, I'm reading LoTR again and it occurred to me that this is a fairly sparkly tree:

<img src="https://upload.wikimedia.org/wikipedia/commons/b/b7/Blason_Gondor.svg" width="150px" />

**Comment by PerilousApricot** [comment]
I've moved the code to a new subpackages `edu.vanderbilt.accre.laurelin`, I'll get upstream updated soon, trying to resolve some layering decisisons now

**Comment by jpivarski** [comment]
Nice name! I think that's a different tree, but perhaps more appropriate.  :)

**Comment by PerilousApricot** [comment]
fixed in 217dc2454a53cdc8f6c86e072624e27b4f58bb1a
